sdk/eventhub/event-hubs/src/eventHubReceiver.ts

```diff
@@ -547,7 +601,78 @@ export class EventHubReceiver extends LinkEntity {
     return result;
   }
 
-  protected _deleteFromCache(): void {
+  /**
+   * Registers the user's onMessage and onError handlers.
+   * Sends buffered events from the queue before adding additional credits to the AMQP link.
+   * @ignore
+   */
+  registerHandlers(
+    onMessage: OnMessage,
+    onError: OnError,
+    maximumCreditCount: number,
+    isStreaming: boolean = false,
+    abortSignal?: AbortSignalLike,
+    onAbort?: OnAbort
+  ): void {
+    this._abortSignal = abortSignal;
+    this._onAbort = onAbort;
+    this._onError = onError;
+    this._onMessage = onMessage;
+    this._isStreaming = isStreaming;
+    // indicate that messages are being received.
+    this._isReceivingMessages = true;
+
+    this._useInternalQueue(onMessage, abortSignal)
+      .then(async (processedEventCount) => {
+        if (this._onMessage !== onMessage) {
+          // the original handler has been removed, so no further action required.
+          return;
+        }
+
+        // check if more messages are required
+        if (!isStreaming && maximumCreditCount - processedEventCount <= 0) {
+          return;
+        }
+
+        if (!this.isOpen()) {
+          try {
+            await this.initialize();
+            if (abortSignal && abortSignal.aborted) {
+              await this.abort();
+            }
+          } catch (err) {
+            return this._onError === onError && onError(err);
+          }
+        } else {
+          log.receiver(
+            "[%s] Receiver link already present, hence reusing it.",
+            this._context.connectionId
+          );
+        }
+        // add credits
+        const existingCredits = this._receiver ? this._receiver.credit : 0;
+        const minimumDefaultCount = isStreaming ? 1 : 0;
+        const creditsToAdd = Math.max(
+          maximumCreditCount - (existingCredits + processedEventCount),
```

Comment:
`processedEventCount` shouldnt matter to streaming receiver when to comes to the credits that need to be added.